### DISCUSSION TITLE
FMODs Future in OF

### DISCUSSION THREAD

**Discussion by NickHardeman** [openframeworks/openFrameworks#8165] [Polls]
What are everyone's thoughts on removing FMOD from the core? It would reduce the dependencies of OF and complexity, ie. submitting to the macOS / iOS app store and copying DLLs, etc.
Seems like one of the last platforms using it by default is Windows. 
There is a Media Foundation sound player that could replace it by defining OF_SOUND_PLAYER_MEDIA_FOUNDATION. Though it would need some thorough testing since it is disabled by default.

FMOD could be moved to a core addon, but not sure if FMOD specific sound player code would reach a wide enough audience to warrant continued support. 

This topic has been discussed over the years, here are a few recent posts:
https://github.com/openframeworks/apothecary/issues/439
https://github.com/openframeworks/openFrameworks/pull/8152

**Comment by dimitre** [comment]
I'm voting move to core addon but it can be a separate addon. or even completely remove.
just a quick way of using it as there are things only possible there. 
I don't remember exactly but I think to have negative arbitrary speeds in audio play.

**Reply by dimitre** [reply]
another thing is FMOD licensing that is incompatible with OF

**Comment by danoli3** [comment]
I think remove Fmod completely as its commercial license required.

We can get better audio with other systems. I recommend Steam Audio and
happy to port it in for all platforms. Valve almost hired me to do so in
the past before I got unwell.

Android I built a low level one for that to be merged fork for Super
Hexagon and that is amazing fast.

Been good diving back in this week. I’ll keep at it

**Comment by ofTheo** [comment]
I was totally in favor of removing it but I think currently VS and msys2 use it by default. 

I think at minimum we should switch the default audio player for Windows VS to the MediaFoundation one in the nightly builds and make sure it runs okay first. 

We are also sort of stuck as msys2 doesn't have the Xaudio2 lib ofMediaFoundationPlayer needs to work. 
So if we remove FMOD msys2 won't have audio. 

Three possible solutions there: 

1. We make OpenAL player default on msys2 
2. We drop msys2 and just focus on VS for Windows keeping things easier to maintain. 
3. We find a workaround for the missing lib ( this could be possible it just didn't seem easy last time @NickHardeman and I looked at it ). 

Anyway - let's make sure all the nightly releases work without fmod - then I'd be totally fine removing it completely as a lib. People could always grab it from 0.12.0 if needed and we could make it a non core addon.

**Reply by ofTheo** [reply]
tag @oxillo @roymacdonald

**Reply by roymacdonald** [reply]
Media Foundation Player is used on windows on ARM64 as default. Not sure what it takes to make it usable on x64. But if it does, I feel that there is a very low hanging fruit.

But, in any case, keep fmod in the nightly builds until it is safe to remove

**Reply by oxillo** [reply]
openAL is already the default soundplayer in msys2 (#6664)

**Reply by ofTheo** [reply]
Thanks @roymacdonald @oxillo - so that's awesome news then. 
I think we can just switch VS to be MF Player and then nothing will be using Fmod by default. 

It does seem like we can get XAudio2 running in msys2 so that could be an optional player or even default if it ends up being fairly stable. @NickHardeman and I can look at that, but that doesn't slow anything down. 

Maybe once we've had a couple of months of no fmod being needed in the nightly builds we can remove?  
Or leave it and remove after OF 13.0

**Comment by ofTheo** [comment]
Also if we did want to switch to Steam Audio it does seem to have pre-built libs for most platforms ( no emscripten ) and I think if the API is C and not C++ it should possibly link with msys2 

<img width="525" alt="image" src="https://github.com/user-attachments/assets/b90df43e-edb8-4bbb-9e97-c5ed0142294d">

https://valvesoftware.github.io/steam-audio/downloads.html

**Comment by roymacdonald** [comment]
I just added a PR to remove fmod from iOS and tvOS. (it was not handling the dylib anymore but still referencing the headers).
I dont have an iOS device to test with. Could please someone test before merging.

**Comment by roymacdonald** [comment]
https://github.com/openframeworks/openFrameworks/pull/8175

**Comment by NickHardeman** [comment]
PR for adding OF_NO_FMOD define in VS so that it uses Media Foundation sound player by default. Changed in the project, so it could be easily reverted by a user if needed
https://github.com/openframeworks/openFrameworks/pull/8180

**Comment by dimitre** [comment]
Time to remove all OF_NO_FMOD defines from the project.
the only functionality I will miss is backwards audio playing with variable speed (not possible on AV) but I can put together an addon if I ever need to use that again
